mlflow/models/resources.py

```diff
@@ -0,0 +1,145 @@
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from dataclasses import dataclass
+from enum import Enum
+from typing import List
+
+
+class ResourceType(Enum):
+    DATABRICKS_VECTOR_SEARCH_INDEX_NAME = "databricks_vector_search_index_name"
+    DATABRICKS_VECTOR_SEARCH_ENDPOINT_NAME = "databricks_vector_search_endpoint_name"
+    DATABRICKS_EMBEDDINGS_ENDPOINT_NAME = "databricks_embeddings_endpoint_name"
+    DATABRICKS_LLM_ENDPOINT_NAME = "databricks_llm_endpoint_name"
+    DATABRICKS_CHAT_ENDPOINT_NAME = "databricks_chat_endpoint_name"
+
+
+class Providers(Enum):
+    DATABRICKS = "databricks"
```

Comment:
I don't think we need this. `target_uri` should be a string. Users should be able to override it, e.g. to specify a databricks profile URI with a different workspace. See https://docs.databricks.com/en/machine-learning/manage-model-lifecycle/multiple-workspaces.html#specify-a-remote-registry

Reply:
In that case, would they need to create different providers to do so?
And use specific class or resources to do so? 
Not sure i am missing something.

Reply:
No, they would just pass the target URI to the constructor of the object. What you're missing is that the resource definition needs a way to specify the Databricks workspace containing the resource; "databricks" is not always sufficient

Reply:
Ahhh interesting.
I was thinking providers would be more of `OpenAI`, `LLAMA_INDEX`, `PINECONE`, etc rather than workspace url or something for example. 

but I see how langchain does it: https://api.python.langchain.com/en/latest/chat_models/langchain_community.chat_models.databricks.ChatDatabricks.html

Updating the code now. :)